hazelcast/src/test/java/com/hazelcast/client/topic/ClientReliableTopicTest.java

```diff
@@ -56,7 +56,7 @@
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
-public class ClientReliableTopicTest extends HazelcastTestSupport {
+public class ClientReliableTopicTest extends TopicTest {
```

Comment:
Since none of the methods in `TopicTest` use the client instance, I don't think it's worth extending it here. If you want to make the tests reusable, we need more involvement on different levels.

First off, it should look like this:
- `ClientReliableTopicTest extends ClientTopicTest` (client-side tests)
- `ReliableTopicAbstractTest extends TopicTest` (member-side tests)

Then, you'd need to go over all of the test methods and make sure they aren't calling something like `instance.getTopic()` or `client.getReliableTopic()` directly in the test. Instead, they would need to call something like `getTopicImplementation().publish()` where `getTopicImplementation` would be some abstract method which is overriden in different tests. For instance:
- `ClientReliableTopicTest` would return `client.getReliableTopic`
- `ReliableTopicAbstractTest` would return `member.getReliableTopic`
- `ClientTopicTest` would return `client.getTopic`
- `TopicTest` would return `member.getTopic`

Personally, I think it's too much involvement for this PR and we can do it in a separate PR. Now, we might just add similar tests on client and member-side, and for both regular topic and ITopic.

Reply:
I've changed it back to `HazelcastTestSupport`.
I can have a deeper look into when this PR is finished. :)